src/models/player.rs

```diff
@@ -1063,6 +1131,227 @@ fn push_seek_to_api<E: Env + 'static>(seek_log_req: SeekLogRequest) -> Effect {
     .into()
 }
 
+fn intro_outro_update<E: Env + 'static>(
+    intro_outro: &mut Option<IntroOutro>,
+    profile: &Profile,
+    selected: Option<&Selected>,
+    video_params: Option<&VideoParams>,
+    series_info: Option<&SeriesInfo>,
+    library_item: Option<&LibraryItem>,
+    skip_gaps: &mut Option<(SkipGapsRequest, Loadable<SkipGapsResponse, CtxError>)>,
+) -> Effects {
+    let skip_gaps_effects = skip_gaps_update::<E>(
+        profile,
+        selected,
+        video_params,
+        series_info,
+        library_item,
+        skip_gaps,
+    );
+
+    let intro_outro_effects = match (skip_gaps, library_item) {
+        (Some((_, Loadable::Ready(response))), Some(library_item)) => {
+            let outro_time = {
+                // durations = Object.keys(intro.gaps || {}).filter(function(d) {
```

Comment:
Those comments containing js code should probably be removed

Reply:
yes, this is here for context on the logic that comes from Stremio.
Will be removed before merging and I've added task for it in the PR.